File: test-procedure-complications.py
#!/usr/bin/env python3
import requests
import json
import sys

url_base = 'http://localhost:7777'
session = requests.Session()

# Step 1: Login
print("Step 1: Logging in...")
login_url = f'{url_base}/site/login'
login_data = {
    'User[username]': 'admin',
    'User[password]': 'admin'
}
try:
    response = session.post(login_url, data=login_data)
    print(f"Login request completed with status: {response.status_code}")
except Exception as e:
    print(f"Error during login: {e}")
    sys.exit(1)

# Step 2: Test the /procedure/complications endpoint without ID
print("\nStep 2: Testing /procedure/complications (no ID)...")
test_url = f'{url_base}/procedure/complications'
try:
    response = session.get(test_url)
    print(f"HTTP Status: {response.status_code}")
    if response.status_code >= 400:
        print(f"ERROR: Page returned {response.status_code}")
        # Check for error messages
        if 'Missing required parameter' in response.text:
            print("Found: Missing required parameter error")
        if 'CHttpException' in response.text:
            print("Found: CHttpException")
        # Extract first error message
        import re
        match = re.search(r'<h1>([^<]+)</h1>', response.text)
        if match:
            print(f"Error Message: {match.group(1)}")
    else:
        print(f"SUCCESS: Page returned {response.status_code}")
except Exception as e:
    print(f"Error: {e}")

# Step 3: Test with a valid procedure ID
print("\nStep 3: Testing /procedure/complications/1 (with ID=1)...")
test_url = f'{url_base}/procedure/complications/1'
try:
    response = session.get(test_url)
    print(f"HTTP Status: {response.status_code}")
    if response.status_code >= 400:
        print(f"ERROR: Page returned {response.status_code}")
    else:
        print(f"SUCCESS: Page returned {response.status_code}")
        print(f"Response length: {len(response.text)} bytes")
        # Try to parse as JSON
        try:
            data = response.json()
            print(f"Response is valid JSON: {type(data)}")
            print(f"Data: {data}")
        except:
            print("Response is not JSON")
except Exception as e:
    print(f"Error: {e}")

File: test-procedure-complications.php
<?php
// Test script to verify the /procedure/complications endpoint

$url = 'http://localhost:7777';
$cookieFile = '/tmp/oe_cookies.txt';

// Step 1: Login
echo "Step 1: Logging in...\n";
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url . '/site/login');
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query([
    'User[username]' => 'admin',
    'User[password]' => 'admin'
]));
curl_setopt($ch, CURLOPT_COOKIEJAR, $cookieFile);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
$response = curl_exec($ch);
curl_close($ch);

// Step 2: Test the /procedure/complications endpoint without ID
echo "Step 2: Testing /procedure/complications (no ID)...\n";
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url . '/procedure/complications');
curl_setopt($ch, CURLOPT_COOKIEFILE, $cookieFile);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HEADER, true);
$response = curl_exec($ch);
$info = curl_getinfo($ch);
curl_close($ch);

echo "HTTP Status: " . $info['http_code'] . "\n";
if ($info['http_code'] >= 400) {
    echo "ERROR: Page returned " . $info['http_code'] . "\n";
    // Extract error message from response
    if (preg_match('/<h1>(.*?)<\/h1>/i', $response, $matches)) {
        echo "Error Message: " . htmlspecialchars($matches[1]) . "\n";
    }
    // Look for "Missing required parameter"
    if (strpos($response, 'Missing required parameter') !== false) {
        echo "Found: Missing required parameter error\n";
    }
    // Look for CHttpException
    if (preg_match('/CHttpException.*?(Missing required parameter.*?\$id)/i', $response, $matches)) {
        echo "Exception Details: " . htmlspecialchars($matches[1]) . "\n";
    }
} else {
    echo "SUCCESS: Page returned " . $info['http_code'] . "\n";
}

// Step 3: Test with a valid procedure ID
echo "\nStep 3: Testing /procedure/complications/1 (with ID=1)...\n";
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url . '/procedure/complications/1');
curl_setopt($ch, CURLOPT_COOKIEFILE, $cookieFile);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HEADER, true);
$response = curl_exec($ch);
$info = curl_getinfo($ch);
curl_close($ch);

echo "HTTP Status: " . $info['http_code'] . "\n";
if ($info['http_code'] >= 400) {
    echo "ERROR: Page returned " . $info['http_code'] . "\n";
} else {
    echo "SUCCESS: Page returned " . $info['http_code'] . "\n";
}

// Clean up
unlink($cookieFile);
?>
